docs/en/Concurrency-Check.md

```diff
@@ -1,3 +1,106 @@
 ## Concurrency Check
 
-TODO
\ No newline at end of file
+### Introduction
+
+Concurrency Check (also known as **Concurrency Control**) refers to specific mechanisms used to ensure data consistency in presence of concurrent changes (multiple processes or users access or change the same data in a database at the same time).
+
+There are two commonly used concurrency control mechanisms/approaches;
+* **Optimistic Locking**: Optimistic Locking is a concurrency control method that does not use record locking. Instead, allows multiple users to attempt to **update** or **delete** the same record without informing the users that others are also attempting to **update/delete** the record. 
+
+    * If a user successfully updates/deletes the record, the other users need to get the latest changes for the current record to be able to make changes. If any other user attempts to change the record without getting the latest state of it, informed that a conflict exists. 
+
+* **Pessimistic Locking**: Pessimistic Locking prevents simultaneous updates to records. As soon as one user starts to edit a record, it is getting locked. Other users who attempt to update this record are informed that another user has an update in progress. The other users must wait until the first user has finished committing their changes, thereby releasing the record lock. Only then other users can make changes to this record.
+
+### Base Classes & Interfaces for Concurrency Stamp
+
+#### `IHasConcurrencyStamp` Interface
+
+It is the base interface for **Concurrency Check** and has a just simple property named `ConcurrencyStamp`. 
+
+```csharp
+public interface IHasConcurrencyStamp 
+{
+    public string ConcurrencyStamp { get; set; }
+}
+```
+
+To be able to use **Concurrency Check** your entity class should inherit from the `IHasConcurrencyStamp` interface directly or indirectly.
+
+#### Base Classes
+
+Most of the **Entity Classes** provided by ABP Framework, inherit from the `IHasConcurrencyStamp` interface. You can see the full list of the base classes:
```

Comment:
Aggregate Root entity classes already implement the `IHasConcurrencyStamp` interface. So, if you are deriving from one of these base classes, you don't need to manually implement the `IHasConcurrencyStamp` interface":